### ISSUE TITLE
GPU underuse

### ISSUE THREAD

**Issue by shanggutianzhen** [scverse/scvi-tools#3176]
Hello, my GPU performance doesn't seem to be fully utilized in my data set
This is my code and my GPU usage at runtime
adata_B = adata[adata.obs['cell_type'] == 'B'].copy()
sc.pp.normalize_total(adata_B)
sc.pp.log1p(adata_B)
adata_B.layers['logcounts'] = adata_B.X.copy()
adata_B.raw = adata_B.copy()
#去批次尝试
batch_key = "sample"
label_key = "sample"
scvi.settings.seed = 39
sc.pp.highly_variable_genes(
    adata_B,
    flavor="seurat_v3",
    n_top_genes=1500,
    layer="counts",
    batch_key="sample",
    subset=True,
    span=0.5
)
scvi.model.SCVI.setup_anndata(adata_B, layer="counts", batch_key="sample")
model = scvi.model.SCVI(adata_B, n_layers=2, n_latent=30, gene_likelihood="nb")
model.train()
path = os.path.join("cellrange", "Bmodel")
model.save(path, overwrite=True)
SCVI_LATENT_KEY = "X_scVI"
adata_B.obsm[SCVI_LATENT_KEY] = model.get_latent_representation()
sc.pp.neighbors(adata_B, use_rep=SCVI_LATENT_KEY)
sc.tl.leiden(adata_B)
sc.tl.umap(adata_B, min_dist=0.3)

+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.120                Driver Version: 550.120        CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090        Off |   00000000:3C:00.0  On |                  Off |
|  0%   40C    P2             77W /  450W |    1406MiB /  24564MiB |     12%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A     57549      C   .../miniconda3/envs/squidpy/bin/python        466MiB |
|    0   N/A  N/A     62867      G   /usr/lib/xorg/Xorg                            303MiB |
|    0   N/A  N/A     63186      G   /usr/bin/gnome-shell                           93MiB |
|    0   N/A  N/A     74138      C   .../miniconda3/envs/squidpy/bin/python        522MiB |
+-----------------------------------------------------------------------------------------+

**Comment by canergen** [comment]
Hi, the models are small in general and parallelization can only happen within one minibatch so for most GPUs there is just not enough parallel compute taking place. You could spend the GPU into larger batch sizes (up to 1024) to speed up training. However, you will need to train for more epochs (look at loss curves).
